### PR TITLE
Latency recording rule

### PR DESCRIPTION
**Pull request by anmolsachan**
ceph-mixins-issue: #38 
Signed-off-by: Anmol Sachan anmol13694@gmail.com

### CODE REVIEW

**Comment by shtripat** [comment]
@anmolsachan Fix the build issue

**Review by shtripat — COMMENTED** [review]
Looks fine and  make sure its tested well

File: rules/rules.libsonnet [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        name: 'ceph.rules',
+        rules: [
+          {
+            record: 'ceph_node_down',
+            expr: |||
+              kube_node_status_condition{condition="Ready",job="kube-state-metrics",status="true"} * on (node) group_right() max(label_replace(ceph_disk_occupation,"node","$1","exported_instance","(.*)")) by (node) == 0
+            ||| % $._config,
+          },
+          {
+            record: 'ceph_aggregate_latency',
```
Comment (umangachapagain) [inline rules/rules.libsonnet]: rename to maybe `cluster:ceph_disk_latency:join_ceph_node_disk_irate1m` based on convention

File: rules/rules.libsonnet [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            expr: |||
+              quantile(.95,(sum((max(label_replace(ceph_disk_occupation,"node","$1","exported_instance","(.*)")) by (node)) * on (node) group_right() (label_replace(max by(pod_ip,node) (kube_pod_info{pod=~"node-exporter.*"}), "instance", "$1:9100", "pod_ip", "(.*)"))  * on (instance) group_right() (irate(node_disk_read_time_seconds_total[1m]) + irate(node_disk_write_time_seconds_total[1m]) /  (irate(node_disk_reads_completed_total[1m]) + irate(node_disk_writes_completed_total[1m]))))))[360m:5m]
```
Comment (umangachapagain) [inline rules/rules.libsonnet]: 1. Why are we using quantile? Isn't this the UI requirement? Can we keep this rule simple?
2. Can we use recording rule to identify ceph disks? This query is too heavy.
3. Can you explain "$1:9100"?
4. Should we be using [360m:5m] ? Isn't this only for UI? What if later we need past 24hrs data?
5. ceph_disk_occupation should have job label.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: rules/rules.libsonnet (hunk 2 above)
Comment (anmolsachan) [reply rules/rules.libsonnet]: > 1. Why are we using quantile? Isn't this the UI requirement? Can we keep this rule simple?
Will remove it. Should be added in UI.
> 2. Can we use recording rule to identify ceph disks? This query is too heavy.
There is no better way to add this query.
> 3. Can you explain "$1:9100"?
To append to the host_ip which comes from : kube_pod_info{pod=~"node-exporter.*"}
> 4. Should we be using [360m:5m] ? Isn't this only for UI? What if later we need past 24hrs data?
Will remove this
> 5. ceph_disk_occupation should have job label.
Will add

**Review by shtripat — APPROVED** [review]
LGTM

**Review by umangachapagain — APPROVED** [review]

**Review by shtripat — APPROVED** [review]

**Pull request merged**